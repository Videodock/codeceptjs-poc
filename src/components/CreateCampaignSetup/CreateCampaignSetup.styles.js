import { createStyles } from '@material-ui/styles';

export default ({ typography, palette, spacing, breakpoints }) => createStyles({
  root: {
    position : 'absolute',
    top      : '50%',
    left     : '50%',
    transform: 'translate(-50%, -50%)',
  },
  container: {
    display   : 'flex',
    alignItems: 'flex-end',
    maxWidth  : 1400,
  },
  content: {
    [breakpoints.down('sm')]: {
      width  : '90vw',
      padding: `0px ${spacing(2)}px`,
    },
    width        : '50vw',
    display      : 'flex',
    flexDirection: 'column',
    alignItems   : 'center',
  },
  form: {
    maxWidth: 590,
    width   : '100%',
  },
  person: {
    [breakpoints.down('sm')]: {
      display: 'none',
    },
    '&:first-child': {
      height: '54vh',
      margin: `0px 24px -56px 24px`,
    },
    '&:last-child': {
      margin: `0px 24px -56px 24px`,
    },
    height: '59vh',
  },
  header: {
    [breakpoints.down('sm')]: {
      fontSize: typography.pxToRem(28),
    },
    fontSize     : typography.pxToRem(37),
    marginBottom : spacing(3),
    textAlign    : 'center',
    letterSpacing: 0.25,
    color        : palette.text.darkPurple,
  },
  inputItem: {
    marginBottom: spacing(2),
  },
  inputHeader: {
    marginBottom: spacing(0.5),
  },
  buttons: {
    display       : 'flex',
    justifyContent: 'flex-end',
  },
});
