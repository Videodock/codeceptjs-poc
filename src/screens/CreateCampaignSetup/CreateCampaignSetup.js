import React from 'react';
import * as PropTypes from 'prop-types';
import CreateCampaignSetupComponent from '../../components/CreateCampaignSetup';

export default function CreateCampaignSetup({
  campaignTitle = '',
  campaignDescription = '',
  campaignDeadline,
  onBackButtonClick,
  onCampaignMetaSubmit,
}) {
  const onFormSubmit = CampaignMetaValues => onCampaignMetaSubmit(CampaignMetaValues);

  return <CreateCampaignSetupComponent onSubmit={onFormSubmit} initialFormValues={{ campaignTitle, campaignDescription, campaignDeadline }} onBackButtonClick={onBackButtonClick} />;
}

CreateCampaignSetup.propTypes = {
  campaignTitle       : PropTypes.string,
  campaignDescription : PropTypes.string,
  campaignDeadline    : PropTypes.object,
  onBackButtonClick   : PropTypes.func,
  onCampaignMetaSubmit: PropTypes.func,
};
