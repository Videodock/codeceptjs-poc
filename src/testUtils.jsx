import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
