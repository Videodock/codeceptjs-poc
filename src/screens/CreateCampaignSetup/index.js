import CreateCampaignSetup from './CreateCampaignSetup';

export * from './CreateCampaignSetup';
export default CreateCampaignSetup;
