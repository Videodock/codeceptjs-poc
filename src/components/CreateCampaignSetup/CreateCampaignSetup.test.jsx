import React from 'react';
import { render } from '../../testUtils';
import CreateCampaignSetup from './CreateCampaignSetup';

describe.skip('<CreateCampaignSetup />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CreateCampaignSetup />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
