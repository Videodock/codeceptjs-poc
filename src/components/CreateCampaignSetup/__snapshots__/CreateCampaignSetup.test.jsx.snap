// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateCampaignSetup /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="makeStyles-container-2"
  >
    <img
      alt="person"
      class="makeStyles-person-5"
      src="personWithBass.png"
    />
    <div
      class="makeStyles-content-3"
    >
      <h1
        class="MuiTypography-root makeStyles-header-6 MuiTypography-h1"
      >
        Letâ€™s get your campaign set up
      </h1>
      <form
        class="makeStyles-form-4"
        id="campaign-meta-form"
      >
        <div
          class="makeStyles-inputItem-7"
        >
          <p
            class="MuiTypography-root makeStyles-inputHeader-8 MuiTypography-body1"
          >
            Give it a catchy title
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Campaign title
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                name="campaignTitle"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="makeStyles-inputItem-7"
        >
          <p
            class="MuiTypography-root makeStyles-inputHeader-8 MuiTypography-body1"
          >
            Explain to your collaborators why you want to improve this score
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Description
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
            >
              <textarea
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
                name="campaignDescription"
                rows="4"
              />
            </div>
          </div>
        </div>
        <div
          class="makeStyles-inputItem-7"
        >
          <p
            class="MuiTypography-root makeStyles-inputHeader-8 MuiTypography-body1"
          >
            When do you need it?
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            format="MMMM DD HH:mm"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
              data-shrink="true"
            >
              Deadline
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
                name="campaignDeadline"
                readonly=""
                type="text"
                value="November 05 17:08"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="makeStyles-buttons-9"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              back
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            form="campaign-meta-form"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Set up my campaign
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
    <img
      alt="person"
      class="makeStyles-person-5"
      src="personTrumpetStandingRight.png"
    />
  </div>
</div>
`;
